src/dune_pkg/archive_driver.ml

```diff
@@ -0,0 +1,106 @@
+open Stdune
+module Process = Dune_engine.Process
+open Fiber.O
+
+module Command = struct
+  type t =
+    { bin : Path.t
+    ; args : archive:Path.t -> target_in_temp:Path.t -> string list
+    }
+end
+
+type t =
+  { command : Command.t Lazy.t
+  ; suffixes : string list
+  }
+
+let which bin_name = Bin.which ~path:(Env_path.path Env.initial) bin_name
+
+let tar_args ~archive ~target_in_temp =
+  [ "xf"; Path.to_string archive; "-C"; Path.to_string target_in_temp ]
+;;
+
+let zip_args ~archive ~target_in_temp =
+  [ Path.to_string archive; "-d"; Path.to_string target_in_temp ]
+;;
+
+let tar =
+  let command =
+    lazy
+      (match
+         (* Test for tar before bsdtar as tar is more likely to be installed
+            and both work equally well for tarballs. *)
+         List.find_map [ "tar"; "bsdtar" ] ~f:which
+       with
+       | Some bin -> { Command.bin; args = tar_args }
+       | None -> Dune_engine.Utils.program_not_found "tar" ~loc:None)
+  in
+  { command; suffixes = [ ".tar"; ".tar.gz"; ".tgz"; ".tar.bz2"; ".tbz" ] }
+;;
+
+let zip =
+  let command =
+    lazy
+      (match which "unzip" with
+       | Some bin -> { Command.bin; args = zip_args }
+       | None ->
+         (* Fall back to using tar to extract zip archives, which is possible in some cases. *)
+         (match
+            (* Test for bsdtar before tar, as if bsdtar is installed then it's
+               likely that the tar binary is GNU tar which can't extract zip
+               archives, whereas bsdtar can. If bsdtar is absent, try using the
+               tar command anyway, as on BSD, MacOS, and Windows systems, the tar
```

Comment:
I think this is only true for FreeBSD and NetBSD, OpenBSD has a very classic `tar`. But close enough and the reasoning is sound I'd say!

Reply:
Classic openbsd. I've updated this comment to not claim that all BSD's come with a tar that can extract zip files.